tests/common/matchers.py

```diff
@@ -32,15 +32,24 @@
 from pyramid import httpexceptions
 
 
-class any_callable(object):  # noqa: N801
+class Matcher(object):
+
+    def __eq__(self, other):
+        raise NotImplementedError('subclasses should provide an __eq__ method')
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
```

Comment:
Is the use of `self.__eq__` rather than `==` a matter of style (to explicitly link it to the method above) or does it serve an additional purpose, like stopping Python trying to invert the arguments?